engine/src/configuration/extended_conf.cc

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2024 Centreon
+ *
+ * This file is part of Centreon Engine.
+ *
+ * Centreon Engine is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * Centreon Engine is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Centreon Engine. If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include "com/centreon/engine/configuration/extended_conf.hh"
+#include "com/centreon/engine/configuration/state.hh"
+#include "com/centreon/exceptions/msg_fmt.hh"
+
+using namespace com::centreon::engine::configuration;
+
+std::list<std::unique_ptr<extended_conf>> extended_conf::_confs;
+
+/**
+ * @brief Construct a new extended state::extended state object
+ *
+ * @param path
+ * @throw exception if json mal formed
+ */
+extended_conf::extended_conf(const std::string& path) : _path(path) {
+  if (::stat(_path.c_str(), &_file_info)) {
+    SPDLOG_LOGGER_ERROR(log_v2::config(), "can't access to {}", _path);
+    throw exceptions::msg_fmt("can't access to {}", _path);
+  }
+  try {
+    _content = common::rapidjson_helper::read_from_file(_path);
+    SPDLOG_LOGGER_INFO(log_v2::config(), "extended conf file {} loaded", _path);
+  } catch (const std::exception& e) {
+    SPDLOG_LOGGER_ERROR(
+        log_v2::config(),
+        "extended_conf::extended_conf : fail to read json content from {}: {}",
+        _path, e.what());
+    throw;
+  }
+}
+
+void extended_conf::reload() {
+  struct stat file_info;
+  if (::stat(_path.c_str(), &file_info)) {
+    SPDLOG_LOGGER_ERROR(log_v2::config(),
+                        "can't access to {} anymore => we keep old content",
+                        _path);
+    return;
+  }
+  if (!memcmp(&file_info, &_file_info, sizeof(struct stat))) {
+    return;
+  }
+  try {
+    _content = common::rapidjson_helper::read_from_file(_path);
+    _file_info = file_info;
+  } catch (const std::exception& e) {
+    SPDLOG_LOGGER_ERROR(log_v2::config(),
+                        "extended_conf::extended_conf : fail to read json "
+                        "content from {} => we keep old content, cause: {}",
+                        _path, e.what());
+  }
+}
+
+void extended_conf::apply_all_to_state(state& dest) {
```

Comment:
Again, an explanation about this function. Do you want to apply the extended_conf to the dest state? Is there another way to apply the extended_conf?
If not, maybe this function could be named `apply(dest)`, like many others in the code...

Reply:
In fact, `_all_` is often a bad key word. We start with *all* and one day there are exceptions...
And a methode `apply()` would apply also all the conf

Reply:
After reading next files, I think something like `update_state(...)` would be better.